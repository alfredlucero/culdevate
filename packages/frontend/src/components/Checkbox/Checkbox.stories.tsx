import React from "react";
import { storiesOf } from "@storybook/react";
import Checkbox from "./index";

const onChange = () => {
  console.log("Change checkbox");
};

storiesOf("Common/Checkbox", module)
  .add("Default Unchecked", () => (
    <Checkbox onChange={onChange} id="checkboxId" checked={false} label="Checkbox Label" />
  ))
  .add("Checked", () => <Checkbox onChange={onChange} id="checkboxId" checked={true} label="Checkbox Label" />)
  .add("Disabled", () => (
    <Checkbox onChange={onChange} id="checkboxId" checked={true} label="Checkbox Label" disabled={true} />
  ));
