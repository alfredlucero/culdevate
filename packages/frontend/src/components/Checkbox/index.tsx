import React from "react";
import cn from "classnames";
import Text from "../Text";
import { CommonProps } from "../commonProps";

export interface CheckboxProps extends CommonProps {
  checked: boolean;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  label: string;
  id: string;
  disabled?: boolean;
}

const Checkbox: React.FC<CheckboxProps> = ({
  checked,
  onChange,
  id,
  label,
  disabled = false,
  testId = "",
  className = "",
  ...passThroughProps
}) => {
  return (
    <div className={cn(className)}>
      <label
        className={cn("inline-flex", "items-center", {
          "cursor-pointer": !disabled,
        })}
        htmlFor={id}
      >
        <input
          type="checkbox"
          checked={checked}
          onChange={onChange}
          disabled={disabled}
          id={id}
          className={cn({
            "cursor-pointer": !disabled,
          })}
          {...(testId !== "" ? { "data-testid": testId } : {})}
          {...passThroughProps}
        />
        <Text
          variant="span"
          className={cn("ml-2", {
            "opacity-75": disabled,
          })}
        >
          {label}
        </Text>
      </label>
    </div>
  );
};

export default Checkbox;
