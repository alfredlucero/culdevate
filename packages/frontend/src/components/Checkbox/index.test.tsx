import React from "react";
import { render } from "@testing-library/react";
import Checkbox from "./index";

describe("<Checkbox />", () => {
  test("should render without error", () => {
    const { container } = render(
      <Checkbox
        onChange={() => {}}
        id="checkboxId"
        checked={false}
        label="Checkbox Label"
        testId="checkboxTestId"
        className="extra-class"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
