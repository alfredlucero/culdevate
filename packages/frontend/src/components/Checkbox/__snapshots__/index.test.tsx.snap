// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render without error 1`] = `
<div>
  <div
    class="extra-class"
  >
    <label
      class="inline-flex items-center cursor-pointer"
      for="checkboxId"
    >
      <input
        class="cursor-pointer"
        data-testid="checkboxTestId"
        id="checkboxId"
        type="checkbox"
      />
      <span
        class="font-sans text-gray-800 text-base ml-2"
      >
        Checkbox Label
      </span>
    </label>
  </div>
</div>
`;
